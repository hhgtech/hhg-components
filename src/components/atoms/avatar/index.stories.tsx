import React from 'react'

import { Story } from '@storybook/react/types-6-0'

import { Avatar, Props } from '.'

export default {
  component: Avatar,
  title: 'Atoms/Avatar',
}

const Template: Story<Props> = (args) => <Avatar {...args} />

export const Square = Template.bind({})
Square.args = {
  avatarImg: <img src="https://source.unsplash.com/random/74x74" />,
}

export const Rounded = Template.bind({})
Rounded.args = {
  avatarImg: <img src="https://source.unsplash.com/random/74x74" />,
  type: 'rounded',
}

export const WithCustomCSS = Template.bind({})
WithCustomCSS.args = {
  avatarImg: <img src="https://source.unsplash.com/random/74x74" />,
  type: 'rounded',
  style: { border: 'solid 1px red' },
}
