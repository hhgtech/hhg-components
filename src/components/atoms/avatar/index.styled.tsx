import styled from 'styled-components'

export const StyledAvatar = styled.div`
  cursor: pointer;
  &[data-avatar-type='square'] {
    img {
      width: 72px;
      height: 72px;
      border-radius: 6px;
    }
  }
  &[data-avatar-type='rounded'] {
    img {
      width: 40px;
      height: 40px;
      border-radius: 50%;
    }
  }
`
