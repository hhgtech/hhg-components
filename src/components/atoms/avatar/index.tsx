import React from 'react'

import { CSSProperties } from 'styled-components'
import { StyledAvatar } from './index.styled'

export type Props = {
  avatarImg: JSX.Element
  onClick: () => void
  type?: 'rounded' | 'square'
  style?: CSSProperties
}

const Avatar = ({ type = 'square', avatarImg, onClick, style }: Props) => {
  return (
    <StyledAvatar
      style={{ ...style }}
      data-avatar-type={type || undefined}
      onClick={onClick}
    >
      {avatarImg}
    </StyledAvatar>
  )
}

export { Avatar }
