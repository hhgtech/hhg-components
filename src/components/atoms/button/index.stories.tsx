import React from 'react'

import { Story } from '@storybook/react/types-6-0'

import { Button, Props } from '.'

export default {
  component: Button,
  title: 'Atoms/Button',
}

const Template: Story<Props> = (args) => <Button {...args}>A Button</Button>

export const Normal = Template.bind({})
Normal.args = {
  color: 'primary',
  size: 'lg',
  isDisabled: false,
  isLoading: false,
}

export const Disabled = Template.bind({})
Disabled.args = {
  color: 'primary',
  size: 'lg',
  isDisabled: true,
  isLoading: false,
}

export const WithIcon = Template.bind({})
WithIcon.args = {
  color: 'primary',
  size: 'lg',
  isDisabled: false,
  isLoading: false,
  icon: <img src="//unsplash.it/16" />,
}

export const Loading = Template.bind({})
Loading.args = {
  color: 'primary',
  size: 'lg',
  isDisabled: false,
  isLoading: true,
}

export const Secondary = Template.bind({})
Secondary.args = {
  color: 'secondary',
  size: 'lg',
  isDisabled: false,
  isLoading: false,
}

export const Medium = Template.bind({})
Medium.args = {
  color: 'primary',
  size: 'md',
  isDisabled: false,
  isLoading: false,
}
