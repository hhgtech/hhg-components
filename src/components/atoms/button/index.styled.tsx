import styled, { keyframes } from 'styled-components'
import { ThemeProps } from 'theme'

export const StyledButton = styled.button`
  width: auto;
  border-radius: 6px;
  cursor: pointer;
  font-weight: ${(props: ThemeProps) => props.theme.sizes.fwBold};
  transition: background-color 0.3s, border-color 0.3s;

  &[data-has-icon] {
    display: flex;
    align-items: center;
    svg {
      width: 16px;
      height: 16px;
      margin-right: 10px;
    }
  }

  &[data-color='primary'] {
    background-color: ${(props: ThemeProps) => props.theme.colors.primaryBase};
    color: ${(props: ThemeProps) => props.theme.colors.white};

    &[data-size='lg'] {
      padding: 12px 24px;
      font-size: 16px;
      line-height: 24px;
    }

    &[data-size='md'] {
      padding: 8px 24px;
      font-size: 14px;
      line-height: 24px;
    }

    &[data-size='sm'] {
      padding: 6px 24px;
      font-size: 13px;
      line-height: 20px;
    }

    &:hover {
      background-color: ${(props: ThemeProps) =>
        props.theme.colors.primaryHover};
    }

    &:active {
      background-color: ${(props: ThemeProps) =>
        props.theme.colors.primaryActive};
    }
  }

  &[data-is-loading] {
    span {
      padding-left: 32px;
    }
    svg {
      position: absolute;
      width: 24px;
      height: 24px;
    }
  }

  &[data-color='secondary'] {
    border: solid 2px ${(props: ThemeProps) => props.theme.colors.primaryBase};
    background-color: ${(props: ThemeProps) => props.theme.colors.white};
    color: ${(props: ThemeProps) => props.theme.colors.primaryBase};

    &[data-size='lg'] {
      padding: 11px 24px;
      font-size: 16px;
      line-height: 24px;
    }

    &[data-size='md'] {
      padding: 7px 24px;
      font-size: 14px;
      line-height: 24px;
    }

    &[data-size='sm'] {
      padding: 5px 24px;
      font-size: 13px;
      line-height: 20px;
    }

    &:hover {
      border-color: ${(props: ThemeProps) => props.theme.colors.primaryHover};
      background-color: ${(props: ThemeProps) =>
        props.theme.colors.primaryHover};
      color: ${(props: ThemeProps) => props.theme.colors.white};
    }

    &:active {
      border-color: ${(props: ThemeProps) => props.theme.colors.primaryActive};
      background-color: ${(props: ThemeProps) =>
        props.theme.colors.primaryActive};
      color: ${(props: ThemeProps) => props.theme.colors.white};
    }
  }

  &:disabled {
    border-color: $gray-300;
    background-color: $gray-300;
    color: ${(props: ThemeProps) => props.theme.colors.white};
    cursor: not-allowed;
  }

  &[data-block] {
    width: 100%;
    padding: 12px auto;
  }
`

const circle = keyframes`
  0% {
        transfrom: rotate(0deg);
      }
      100% {
        transform: rotate(360deg);
      }
`

export const StyledLoader = styled.svg`
  animation: ${circle} 1.2s linear infinite;
`
