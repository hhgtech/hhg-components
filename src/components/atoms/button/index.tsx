import React, { ReactNode } from 'react'

import { StyledButton, StyledLoader } from './index.styled'

export type Props = {
  size: 'lg' | 'md' | 'sm'
  children: ReactNode
  color?: 'primary' | 'secondary'
  isDisabled?: boolean
  isBlock?: boolean
  title?: string
  className?: string
  icon?: JSX.Element
  isLoading?: boolean
  loadingText?: string
  onClick?: () => void
}

const Button = ({
  size,
  color = 'primary',
  isDisabled,
  isBlock,
  title,
  className,
  onClick,
  icon,
  children,
  isLoading,
  loadingText,
}: Props) => (
  <StyledButton
    className={className}
    data-size={size}
    data-color={color}
    data-block={isBlock}
    data-has-icon={!!icon || undefined}
    disabled={isDisabled}
    onClick={onClick}
    title={title}
    data-is-loading={isLoading || undefined}
  >
    {isLoading ? (
      <div>
        {color === 'secondary' && (
          <>
            <svg
              width="24"
              height="24"
              fill="none"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle cx="12" cy="12" r="10" stroke="#DBDFE5" strokeWidth="4" />
            </svg>

            <StyledLoader
              width="24"
              height="24"
              fill="none"
              xmlns="http://www.w3.org/2000/svg"
            >
              <mask id="a" fill="#fff">
                <path d="M12 0A12 12 0 11.19 9.88l3.9.7A8.03 8.03 0 1012 3.98V0z" />
              </mask>
              <path
                d="M12 0A12 12 0 11.19 9.88l3.9.7A8.03 8.03 0 1012 3.98V0z"
                stroke="#2D87F3"
                strokeWidth="8"
                mask="url(#a)"
              />
            </StyledLoader>
          </>
        )}

        {color === 'primary' && (
          <>
            <svg
              width="24"
              height="24"
              viewBox="0 0 24 24"
              fill="none"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle cx="12" cy="12" r="10" stroke="white" strokeWidth="4" />
            </svg>
            <StyledLoader
              width="24"
              height="24"
              viewBox="0 0 24 24"
              fill="none"
            >
              <mask id="path-1-inside-1" fill="white">
                <path d="M12 0C14.464 2.93833e-08 16.8683 0.758527 18.8863 2.17253C20.9042 3.58653 22.438 5.58746 23.2791 7.90347C24.1203 10.2195 24.2281 12.7383 23.5879 15.1177C22.9477 17.4971 21.5906 19.6218 19.7009 21.2031C17.8111 22.7843 15.4804 23.7456 13.0254 23.9561C10.5704 24.1667 8.11007 23.6164 5.97869 22.38C3.8473 21.1436 2.14819 19.281 1.1122 17.0454C0.0761991 14.8097 -0.246461 12.3093 0.188053 9.8839L4.09856 10.5845C3.8079 12.2069 4.02374 13.8795 4.71676 15.375C5.40977 16.8705 6.54637 18.1165 7.97213 18.9435C9.39789 19.7706 11.0437 20.1387 12.6859 19.9979C14.3282 19.857 15.8873 19.214 17.1514 18.1563C18.4155 17.0985 19.3233 15.6772 19.7516 14.0856C20.1798 12.4939 20.1077 10.8089 19.545 9.25969C18.9823 7.71042 17.9563 6.37193 16.6065 5.42605C15.2566 4.48017 13.6483 3.97277 12 3.97277L12 0Z" />
              </mask>
              <path
                d="M12 0C14.464 2.93833e-08 16.8683 0.758527 18.8863 2.17253C20.9042 3.58653 22.438 5.58746 23.2791 7.90347C24.1203 10.2195 24.2281 12.7383 23.5879 15.1177C22.9477 17.4971 21.5906 19.6218 19.7009 21.2031C17.8111 22.7843 15.4804 23.7456 13.0254 23.9561C10.5704 24.1667 8.11007 23.6164 5.97869 22.38C3.8473 21.1436 2.14819 19.281 1.1122 17.0454C0.0761991 14.8097 -0.246461 12.3093 0.188053 9.8839L4.09856 10.5845C3.8079 12.2069 4.02374 13.8795 4.71676 15.375C5.40977 16.8705 6.54637 18.1165 7.97213 18.9435C9.39789 19.7706 11.0437 20.1387 12.6859 19.9979C14.3282 19.857 15.8873 19.214 17.1514 18.1563C18.4155 17.0985 19.3233 15.6772 19.7516 14.0856C20.1798 12.4939 20.1077 10.8089 19.545 9.25969C18.9823 7.71042 17.9563 6.37193 16.6065 5.42605C15.2566 4.48017 13.6483 3.97277 12 3.97277L12 0Z"
                stroke="#91CAFF"
                strokeWidth="8"
                mask="url(#path-1-inside-1)"
              />
            </StyledLoader>
          </>
        )}

        <span>{loadingText}</span>
      </div>
    ) : (
      <span>
        {icon && !isLoading && icon} {children}
      </span>
    )}
  </StyledButton>
)

export { Button }
