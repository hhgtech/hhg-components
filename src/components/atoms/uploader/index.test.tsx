import React from 'react'
import { render, cleanup } from '@testing-library/react'
import 'jest-styled-components'

import { Uploader } from '.'

import MockTheme from '../../../mockTheme'
import { Avatar } from '../avatar'
import { Button } from '../button'

afterEach(cleanup)

const icon = (
  <svg
    id="prefix__Layer_1"
    data-name="Layer 1"
    xmlns="http://www.w3.org/2000/svg"
    viewBox="0 0 15.5 12.5"
  >
    <defs>
      <style>
        {
          '.prefix__cls-1{fill:none;stroke:#262626;stroke-linecap:round;stroke-linejoin:round;stroke-width:1.5px}'
        }
      </style>
    </defs>
    <title>{'camera'}</title>
    <path
      className="prefix__cls-1"
      d="M15 11.78a1.19 1.19 0 01-.37.86 1.31 1.31 0 01-.9.36H2.27a1.31 1.31 0 01-.9-.36 1.19 1.19 0 01-.37-.86V5.06a1.19 1.19 0 01.37-.87 1.31 1.31 0 01.9-.36h2.55L6.09 2h3.82l1.27 1.83h2.55a1.31 1.31 0 01.9.36 1.19 1.19 0 01.37.87z"
      transform="translate(-.25 -1.25)"
    />
    <path
      className="prefix__cls-1"
      d="M8 10a2 2 0 10-2-2 2 2 0 002 2z"
      transform="translate(-.25 -1.25)"
    />
  </svg>
)

const avatar = (
  <Avatar
    type="square"
    avatarImg={<img src="https://source.unsplash.com/random/74x74" />}
    onClick={() => {}}
  />
)

const button = (
  <Button size="md" color="secondary" className="default-button" icon={icon}>
    <svg
      id="prefix__Layer_1"
      data-name="Layer 1"
      xmlns="http://www.w3.org/2000/svg"
      viewBox="0 0 13.5 13.5"
    >
      <defs>
        <style>
          {
            '.prefix__cls-1{fill:none;stroke:#000;stroke-linecap:round;stroke-linejoin:round;stroke-width:1.5px}'
          }
        </style>
      </defs>
      <title>{'upload'}</title>
      <path
        className="prefix__cls-1"
        d="M14 10v2.67A1.33 1.33 0 0112.67 14H3.33A1.33 1.33 0 012 12.67V10M11.33 5.33L8 2 4.67 5.33M8 2v8"
        transform="translate(-1.25 -1.25)"
      />
    </svg>
    {'Upload Image 2'}
  </Button>
)

let type: 'badge' | 'button' = 'badge'

test('it renders', () => {
  const uploader = render(
    <MockTheme>
      <Uploader
        type={type}
        onFileSelected={(files: FileList) => {
          console.log(files)
        }}
        withIcon={icon}
        inputId={'1'}
        fileAcceptType="image/*"
        multiple={true}
        withElement={type === 'badge' ? avatar : button}
      />
    </MockTheme>,
  )

  const self = uploader.container.firstChild
  expect(self).toMatchSnapshot()
})
