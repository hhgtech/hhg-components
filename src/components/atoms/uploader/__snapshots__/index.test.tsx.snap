// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders 1`] = `
.c0 .default-button[data-color='secondary'] {
  border: solid 2px #bfbfbf;
  color: #262626;
}

.c0 .default-button[data-color='secondary']:hover {
  border: solid 2px #bfbfbf;
  background-color: #bfbfbf;
  color: #fff;
}

.c0 .default-button[data-color='secondary']:hover > svg {
  fill: #fff;
}

.c1 {
  display: inline-block;
  position: relative;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 32px;
  height: 32px;
  border-radius: 50%;
  background-color: #f2f2f2;
  position: absolute;
  bottom: -4px;
  right: -4px;
}

.c3 > svg {
  height: 16px;
  width: 16px;
}

.c2 {
  cursor: pointer;
}

.c2[data-avatar-type='square'] img {
  width: 72px;
  height: 72px;
  border-radius: 6px;
}

.c2[data-avatar-type='rounded'] img {
  width: 40px;
  height: 40px;
  border-radius: 50%;
}

<div
  class="c0"
>
  <div
    class="c1"
    data-is-attached-to-element="true"
  >
    <div
      class="c2"
      data-avatar-type="square"
    >
      <img
        src="https://source.unsplash.com/random/74x74"
      />
    </div>
    <div
      class="c3"
      data-is-attached-to-element="true"
    >
      <svg
        data-name="Layer 1"
        id="prefix__Layer_1"
        viewBox="0 0 15.5 12.5"
        xmlns="http://www.w3.org/2000/svg"
      >
        <defs>
          <style>
            .prefix__cls-1{fill:none;stroke:#262626;stroke-linecap:round;stroke-linejoin:round;stroke-width:1.5px}
          </style>
        </defs>
        <title>
          camera
        </title>
        <path
          class="prefix__cls-1"
          d="M15 11.78a1.19 1.19 0 01-.37.86 1.31 1.31 0 01-.9.36H2.27a1.31 1.31 0 01-.9-.36 1.19 1.19 0 01-.37-.86V5.06a1.19 1.19 0 01.37-.87 1.31 1.31 0 01.9-.36h2.55L6.09 2h3.82l1.27 1.83h2.55a1.31 1.31 0 01.9.36 1.19 1.19 0 01.37.87z"
          transform="translate(-.25 -1.25)"
        />
        <path
          class="prefix__cls-1"
          d="M8 10a2 2 0 10-2-2 2 2 0 002 2z"
          transform="translate(-.25 -1.25)"
        />
      </svg>
    </div>
  </div>
  <input
    accept="image/*"
    id="uploader-input-1"
    multiple=""
    style="display: none;"
    type="file"
  />
</div>
`;
